### PR TITLE
add RELEASE_NOTE generation script

### PR DESCRIPTION
**Pull request by tkxkd0159**
## Description

closes: #XXXX

* Generate `RELEASE_NOTE` by code
* Make `what's changed` only for regular release, not for release candidates. Because people mostly want to know the difference between main release. If we deploy the rc release notes separately, It's hard to track the changes at once. 
* Remove `RELEASE_CHANGELOG`. Use `CHANGELOG` instead.

## How has this been tested?

Sample output
```
# Finschia v3.0.0 Release Note

## Highlights
* patch [CWA-2023-004 issue](https://forum.cosmos.network/t/high-severity-security-patch-upcoming-on-wed-10th-cwa-2023-004-brought-to-you-by-certik-and-confio/12840)
* Ensure smart contracts compiled with Rust v1.70 run without errors.
* disable custom querier in wasm

## What's Changed
Check out all the changes [here](https://github.com/Finschia/finschia/compare/v2.0.2...v3.0.0)

### Features
* (build) [\#329](https://github.com/Finschia/finschia/pull/329) rename namespace to v3

### Improvements
* (improvements) [\#230](https://github.com/Finschia/finschia/pull/230) fix Makefile for format and execute make format #230
* (chore) [\#299](https://github.com/Finschia/finschia/pull/299) remove x/token and x/collection apis in swagger

### Bug Fixes
* (finschia-sdk) [\#297](https://github.com/Finschia/finschia/pull/297) bump up finschia-sdk from v0.48.0 to v0.48.1
* (ostracon) [\#333](https://github.com/Finschia/finschia/pull/333) bump up ostracon from v1.1.2 to v1.1.3

### Breaking Changes
* (wasmd) [\#328](https://github.com/Finschia/finschia/pull/328) bump up wasmd from v0.2.0 to v0.3.0

### Build, CI
* (ci) [\#290](https://github.com/Finschia/finschia/pull/290) remove autopr ci
* (ci) [\#291](https://github.com/Finschia/finschia/pull/291) fix goreleaser ci error and replace release-build
* (repo) [\#295](https://github.com/Finschia/finschia/pull/295) setup CODEOWNERS and backport action
* (ci) [\#296](https://github.com/Finschia/finschia/pull/296) bump actions/checkout from 3 to 4
* (ci) [\#305](https://github.com/Finschia/finschia/pull/305) add e2e-ibc ci
* (build) [\#316](https://github.com/Finschia/finschia/pull/316) change docker image version to fix build error

## Base sub modules
* Ostracon: [v1.1.3](https://github.com/Finschia/ostracon/tree/v1.1.3)
* Finschia-sdk: [v0.48.1](https://github.com/Finschia/finschia-sdk/tree/v0.48.1)
* Finschia/wasmd: [v0.3.0](https://github.com/Finschia/wasmd/tree/v0.3.0)
* Finschia/ibc-go: [v4.3.1](https://github.com/Finschia/ibc-go/tree/v4.3.1)

## Build from source
You must use Go 1.20 if building from source
\```shell
git clone https://github.com/Finschia/finschia
cd finschia && git checkout v3.0.0
make install
\```

## Run with Docker
If you want to run fnsad in a Docker container, you can use the Docker images.
* docker image: `finschia/finschianode:3.0.0`
\```shell
docker run finschia/finschianode:3.0.0 fnsad version
# 3.0.0
\```

## Download binaries

Binaries for linux and darwin are available below.

```

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/Finschia/finschia/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/Finschia/finschia/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [ ] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by zemyblue — COMMENTED** [review]
Since this release note is also used as the URL for a software update proposal, it should be written in a language that regular Holders, not developers, can understand.

I think it would be good to use it as a script to create a basic release note that can be processed. In the case of a very simple minor update, it may be possible to use it without change.

However, it seems that in order to edit RELEASE_NOTE.md, it had to be modified in a separate PR. Is this possible by editing tag.yml? I don't want it to be automatically recorded as a release note when a tag is added. Is there a way?

**Comment by tkxkd0159** [comment]
> However, it seems that in order to edit RELEASE_NOTE.md, it had to be modified in a separate PR. Is this possible by editing tag.yml? I don't want it to be automatically recorded as a release note when a tag is added. Is there a way?

This Github release note will be generated when we push the tag. We have to take responsibility to maintain our `RELEASE_CHANGELOG` well and shouldn't push the tag if it's not done yet. And RELEASE_NOTE will be used as template file from this PR. We should fix the content directly in Github if we need it.

File: RELEASE_CHANGELOG.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 
 # Changelog
 
-## [v3.0.0] - 2024-03-04
```
Comment (zemyblue) [inline RELEASE_CHANGELOG.md:37]: Why did you remove this update date?

File: RELEASE_CHANGELOG.md [hunk 2]
```diff
@@ -264,26 +264,3 @@ This version based on finschia-sdk v0.46.0-rc2
...
-[v2.0.0]: https://github.com/Finschia/finschia/releases/tag/v2.0.0
-[v1.0.0]: https://github.com/Finschia/finschia/releases/tag/v1.0.0
-[v0.7.0]: https://github.com/Finschia/finschia/releases/tag/v0.7.0
-[v0.6.0]: https://github.com/Finschia/finschia/releases/tag/v0.6.0
-[v0.5.0]: https://github.com/Finschia/finschia/releases/tag/v0.5.0
-[v0.4.0]: https://github.com/Finschia/finschia/releases/tag/v0.4.0
-[v0.3.0]: https://github.com/Finschia/finschia/releases/tag/v0.3.0
-[v0.2.0]: https://github.com/Finschia/finschia/releases/tag/v0.2.0
-[v0.1.0]: https://github.com/Finschia/finschia/releases/tag/v0.1.0
-[gaia v4.0.4]: https://github.com/cosmos/gaia/releases/tag/v4.0.4
```
Comment (zemyblue) [inline RELEASE_CHANGELOG.md:289]: Why did you remove this realease links and gaia update changelog?

Comment (tkxkd0159) [reply RELEASE_CHANGELOG.md:289]: Why do we need to leave Gaia release? becasue it's Finschia's release changelog. But I will revert our release link.

File: RELEASE_CHANGELOG.md (hunk 1 above)
Comment (tkxkd0159) [reply RELEASE_CHANGELOG.md:37]: Some release version (below v0.7) doesn't have date. So I unified the subject style for parsing. I thought we can find the date at release note or commit. If you need to maintain the date format, we need to change previous release title.

File: RELEASE_CHANGELOG.md (hunk 2 above)
Comment (zemyblue) [reply RELEASE_CHANGELOG.md]: Because finschia was forked from gaia 4.0.4.

File: RELEASE_CHANGELOG.md (hunk 1 above)
Comment (zemyblue) [reply RELEASE_CHANGELOG.md:37]: We did not release as public below v0.7. And we did not count it before. But we counted the release date after public. So how about remaining the histories?

File: CHANGELOG.md [hunk 3]
```diff
@@ -48,5 +48,235 @@ Ref: https://keepachangelog.com/en/1.0.0/
...
+[v2.0.0]: https://github.com/Finschia/finschia/releases/tag/v2.0.0
+[v1.0.0]: https://github.com/Finschia/finschia/releases/tag/v1.0.0
+[v0.7.0]: https://github.com/Finschia/finschia/releases/tag/v0.7.0
+[v0.6.0]: https://github.com/Finschia/finschia/releases/tag/v0.6.0
+[v0.5.0]: https://github.com/Finschia/finschia/releases/tag/v0.5.0
+[v0.4.0]: https://github.com/Finschia/finschia/releases/tag/v0.4.0
+[v0.3.0]: https://github.com/Finschia/finschia/releases/tag/v0.3.0
+[v0.2.0]: https://github.com/Finschia/finschia/releases/tag/v0.2.0
+[v0.1.0]: https://github.com/Finschia/finschia/releases/tag/v0.1.0
+[gaia v4.0.4]: https://github.com/cosmos/gaia/releases/tag/v4.0.4
```
Comment (zemyblue) [inline CHANGELOG.md:282]: This release link is not used in this file. So I think it can be removed. 
But please add original fork information below the v0.1.0 description

This release link is deprecated in this file. So I think it can be removed.
But please add at least some original fork information under the v0.1.0 description.
Like “This registory forked [gaia v4.0.4.](https://github.com/cosmos/gaia/releases/tag/v4.0.4)”

File: .github/workflows/tag.yml [hunk 4]
```diff
@@ -8,7 +8,7 @@ on:
   push:
     # Sequence of patterns matched against refs/tags
     tags:
-      - "v[0-9]+.[0-9]+.[0-9]+*" # Push events to matching v*, i.e. v1.0, v20.15.10
+      - "v[0-9]+.[0-9]+.[0-9]+" # Push events to matching v*, i.e. v1.0, v20.15.10
```
Comment (zemyblue) [inline .github/workflows/tag.yml:11]: Can this method be used in cases such as `v1.0.0-rc1`?

File: CHANGELOG.md (hunk 3 above)
Comment (tkxkd0159) [reply CHANGELOG.md:282]: Yeah you can check at `RELEASE_DESCR`

File: .github/workflows/tag.yml (hunk 4 above)
Comment (tkxkd0159) [reply .github/workflows/tag.yml:11]: Please check the second explanation of PR.

Comment (zemyblue) [reply .github/workflows/tag.yml:11]: If then, we cannot use release note and assets in rc* tags? Oh, it is a problem.

Comment (tkxkd0159) [reply .github/workflows/tag.yml]: Then I will add it again.

**Review by ulbqb — APPROVED** [review]
